retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.ts

```diff
@@ -0,0 +1,68 @@
+import { InvestorsRepository } from './../../../domain/investors/investor.repository';
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Investors } from 'src/app/domain/investors/models/investor';
+
+@Component({
+  selector: 'app-pop-up',
+  templateUrl: './pop-up.component.html',
+})
+export class PopUpInvestorComponent implements OnInit {
+  popupInvestorForm!: FormGroup;
+  allInvestors: Investors[] = [];
+  submitted: boolean = false;
+  constructor(
+    private formBuilder: FormBuilder,
+    private investorsRepository: InvestorsRepository,
+    public dialogRef: MatDialogRef<PopUpInvestorComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: Investors
+  ) {}
+  ngOnInit(): void {
+    this.investorsForm();
+    this.fetchData(this.data);
+  }
+  investorsForm() {
+    this.popupInvestorForm = this.formBuilder.group({
+      id: [''],
+      fullName: ['', [Validators.required]],
+      nickName: ['', [Validators.required]],
+      nationalId: ['', [Validators.required]],
+      primaryPhoneNo: ['', [Validators.required]],
+      secondaryPhoneNo: ['', [Validators.required]],
+      address: ['', [Validators.required]],
+      investorType: ['', [Validators.required]],
+      balance: ['', [Validators.required]],
+      startDate: ['', [Validators.required]],
```

Comment:
why you make all controls required depending on what?
and 
phone must be 11  number 
email like emails 
nationalId must be 14 number

Reply:
what should be required, and what should not?

Reply:
in general, from swagger object 
but now it not handled yet
so you can ask for that before make your that


Temporarily, you can make investorType, fullName, nationalId and primaryPhoneNo required